ludwig/visualize.py

```diff
@@ -3710,7 +3710,8 @@ def hyperopt_hiplot(hyperopt_stats_path, output_directory=None, **kwargs):
 
 def hyperopt_results_to_dataframe(hyperopt_results, hyperopt_parameters, metric):
     df = pd.DataFrame([{metric: res["metric_score"], **res["parameters"]} for res in hyperopt_results])
-    df = df.astype({hp_name: hp_params[TYPE] for hp_name, hp_params in hyperopt_parameters.items()})
+    # TODO: work out how to use SPACE key instead of TYPE key
+    # df = df.astype({hp_name: hp_params[SPACE] for hp_name, hp_params in hyperopt_parameters.items()})
```

Comment:
We can use a sort of inverse mapping of RAY_TUNE_FLOAT_SPACES and the like, Space -> type ?

Reply:
define function `visualize._convert_space_to_dtype()` to translate from SPACE to dtype.